flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -1244,4 +1278,99 @@ private static List<RowData> fetchAllResults(ResultFetcher resultFetcher) {
         }
         return results;
     }
+
+    private JobExecutionResult executeJob(
+            String script,
+            Configuration executionConfig,
+            OperationExecutor operationExecutor,
+            OperationHandle operationHandle) {
+        if (isApplicationMode(operationExecutor.getSessionContext().getSessionConf())) {
+            return executeApplicationJob(script, executionConfig, operationExecutor);
+        } else {
+            return executeSessionJob(script, executionConfig, operationExecutor, operationHandle);
+        }
+    }
+
+    private boolean isApplicationMode(Configuration configuration) {
+        String target = configuration.get(TARGET);
+        return target.endsWith("application");
+    }
+
+    private JobExecutionResult executeSessionJob(
+            String script,
+            Configuration executionConfig,
+            OperationExecutor operationExecutor,
+            OperationHandle operationHandle) {
+        Configuration mergedConfig =
+                new Configuration(operationExecutor.getSessionContext().getSessionConf());
```

Comment:
We don't need to merge the sessionConf with executionConfig here, it will be merged when calling the method `operationExecutor.executeStatement`.